contracts/CompoundMiddleContract.sol

```diff
@@ -88,14 +88,18 @@ contract CompoundMiddleContract {
      * @param _comptrollerAddress address of the comptroller contract in Compound
      * @param _cTokenAddress address of the cToken contract in Compound
      * @param _amountToBorrowInWei amount of ether to be borrowed in wei
+     * @param _borrowType a bool denoting whether the borrow function was called by the Leverage contract or the user
+     * @param _leverageContractAddress address of the Leverage contract (can be address(0) when _borrowType is false)
      * @return uint256 the current borrow balance of the user
      */
     function borrowEth(
         address payable _cEtherAddress,
         address _comptrollerAddress,
         address _cTokenAddress,
-        uint256 _amountToBorrowInWei
-    ) public returns (uint256) {
+        uint256 _amountToBorrowInWei,
+        bool _borrowType, 
```

Comment:
This is what I am talking about, Let's not use this _borrowType.
Just have leverage as a separate feature!